code/modules/power/gourmonger.dm

```diff
@@ -0,0 +1,276 @@
+#define GOURMONGER_METABOLISM 1
+#define GOURMONGER_STARVING 50
+#define GOURMONGER_SATISFIED 500
+#define TOO_MANY_GOURS 30
+
+var/global/gourmonger_saturation = 0
+
+/mob/living/simple_animal/hostile/gourmonger
+	name = "Gourmonger"
+	desc = "A bio-engineered lifeform designed to recycle excess organic matter into fuel."
+	icon = 'icons/mob/gourmonger.dmi'
+	icon_state = "gourmonger"
+	icon_living = "gourmonger"
+	icon_dead = "gourmonger"	//This shouldn't happen
+	faction = "gourmonger"
+	health = 125
+	maxHealth = 125
+	melee_damage_lower = 5
+	melee_damage_upper = 10	//The rads are the real danger
+	stat_attack = 2
+	search_objects = 1
+	environment_smash_flags = SMASH_CONTAINERS
+	wanted_objects = list(/obj/item/weapon/reagent_containers/food/snacks)
+	min_oxy = 0	//I dunno it breathes food or something
+	max_co2 = 0
+	var/hangry = FALSE
+	var/kcalPower = 75
+	var/growToSplit = 15
+	var/currentlyMunching = FALSE
+	var/mob/living/sniffTarget = null	//because using hostile's target var led to too many issues
+
+/mob/living/simple_animal/hostile/gourmonger/New()
+	..()
+	gourmonger_saturation++
+	growToSplit += rand(0, 15)	//For funsies
+	kcalPower += rand(0, 50)
+
+/mob/living/simple_animal/hostile/gourmonger/Life()
+	if(..())
+		if(kcalPower > 0)
+			kcalPower -= GOURMONGER_METABOLISM
+		radBurst(kcalPower/3)
+		hungerCheck()
+		if(hangry && !target)
+			if(sniffTarget)
+				chargeToPrey()
+			else
+				findPrey()
+
+/mob/living/simple_animal/hostile/gourmonger/death()
+	gourmonger_saturation--
+	radBurst(kcalPower*3)
+	divideMeat()
+	gibs(get_turf(src))
+	..()
+	qdel(src)
+
+
+/mob/living/simple_animal/hostile/gourmonger/CanAttack(var/atom/the_target) //My hands are tied. I swear I had a reason for this copy paste. It also trims some weight from targeting.
+	if(currentlyMunching)
+		return 0
+	if(the_target.invisibility)
+		return 0
+	if(isliving(the_target))
+		var/mob/living/L = the_target
+		if(L.flags & INVULNERABLE)
+			return 0
+		if(isMoMMI(L))
+			return 0
+		if(L.faction == src.faction || faction == "\ref[L]")
+			return 0
+		if(faction == "slimesummon")
+			if(isslime(L))
+				return 0
+			if(ishuman(L))
+				var/mob/living/carbon/human/H = L
+				if((isslimeperson(H)) || (isgolem(H)))
+					return 0
+		if((istype(L,/mob/living/simple_animal/corgi/Ian) || istype(L,/mob/living/carbon/human/dummy)) && (faction == "adminbus mob"))
+			return 0
+		if(friends.Find(L))
+			return 0
+		if(!L.stat && !hangry)	//So we attack corpses but not living creatures unless we're starving
+			return 0
+		if(sniffTarget)
+			sniffTarget = null
+		return 1
+	if(isobj(the_target))
+		if(is_type_in_list(the_target, wanted_objects))
+			if(sniffTarget)
+				sniffTarget = null
+			return 1
+	return 0
+
+/mob/living/simple_animal/hostile/gourmonger/Goto(var/target, var/delay, var/minimum_distance)
+	if(hangry)
+		chargeToPrey(target)
+	..()
+
+/mob/living/simple_animal/hostile/gourmonger/proc/hungerCheck()
+	if(!hangry && kcalPower < GOURMONGER_STARVING)
```

Comment:
Collapse this is into `hangry = (kcalPower < GOURMONGER_STARVING)`

Reply:
That works? Learned something.

Reply:
Wait, wouldn't that cause him to stop being hangry if it's above the starving threshold? There's two thresholds: 50 to become hangry and 500 to stop being hangry. Otherwise it's too easy to make him stop.

Reply:
It would, yes. He probably misread what was going on. You can change the next `if` to `else if` though since the two conditions are mutually exclusive.